worldedit_commands/cuboid.lua

```diff
@@ -0,0 +1,205 @@
+minetest.register_chatcommand("/outset", {
+	params = "<amount> [h|v]",
+	description = "outset the selection",
+	privs = {worldedit=true},
+	func = function(name, param)
+		local find, _, amount, dir = param:find("^(%d+)[%s+]?([hv]?)$")
+		
+		if find == nil then
+			return false, "invalid usage: " .. param
+		end
+		
+		local pos1 = worldedit.pos1[name]
+		local pos2 = worldedit.pos2[name]
+		
+		if pos1 == nil or pos2 == nil then
+			return false, 
+				"Undefined region. Region must be defined beforehand."
+		end
+		
+		if dir == "" then
+			assert(worldedit.cuboid_volumetricexpand(name, amount))
+		elseif dir == "h" then
+			assert(worldedit.cuboid_linealexpand(name, 'x', 1, amount))
+			assert(worldedit.cuboid_linealexpand(name, 'x', -1, amount))
+			assert(worldedit.cuboid_linealexpand(name, 'z', 1, amount))
+			assert(worldedit.cuboid_linealexpand(name, 'z', -1, amount))
+		elseif dir == "v" then
+			assert(worldedit.cuboid_linealexpand(name, 'y', 1, amount))
+			assert(worldedit.cuboid_linealexpand(name, 'y', -1, amount))
+		else
+			return false, "Unknown error"
+		end
+		
+		worldedit.marker_update(name)
+		return true, "Region outset by " .. amount .. " blocks"
+      end,
+  }
+)
+
+minetest.register_chatcommand("/inset", {
+	params = "<amount> [h|v]",
+	description = "inset the selection",
+	privs = {worldedit=true},
+	func = function(name, param)
+		local find, _, amount, dir = param:find("^(%d+)[%s+]?([hv]?)$")
+		
+		if find == nil then
+			return false, "invalid usage: " .. param
+		end
+		
+		local pos1 = worldedit.pos1[name]
+		local pos2 = worldedit.pos2[name]
+		
+		if pos1 == nil or pos2 == nil then
+			return false, 
+				"Undefined region. Region must be defined beforehand."
+		end
+		
+		if dir == "" then
+			assert(worldedit.cuboid_volumetricexpand(name, -amount))
+		elseif dir == "h" then
+			assert(worldedit.cuboid_linealexpand(name, 'x', 1, -amount))
+			assert(worldedit.cuboid_linealexpand(name, 'x', -1, -amount))
+			assert(worldedit.cuboid_linealexpand(name, 'z', 1, -amount))
+			assert(worldedit.cuboid_linealexpand(name, 'z', -1, -amount))
+		elseif dir == "v" then
+			assert(worldedit.cuboid_linealexpand(name, 'y', 1, -amount))
+			assert(worldedit.cuboid_linealexpand(name, 'y', -1, -amount))
+		else
+			return false, "Unknown error"
+		end
+		
+		worldedit.marker_update(name)
+		return true, "Region inset by " .. amount .. " blocks"
+      end,
+  }
+)
+
+
+minetest.register_chatcommand("/shift", {
+	params = "<amount> [up|down|left|right|front|back]",
+	description = "Moves the selection region. Does not move contents.",
+	privs = {worldedit=true},
+	func = function(name, param)
+		local pos1 = worldedit.pos1[name]
+		local pos2 = worldedit.pos2[name]
+		local find, _, amount, direction = param:find("(%d+)%s*(%l*)")
```

Comment:
* Negative amounts should work here, add `-?` before `%d`.
  * This is still in the wrong order.